diff --git a/spacy/lang/grc/test_text.py b/spacy/lang/grc/test_text.py
deleted file mode 100644
index 5d8317c36..000000000
--- a/spacy/lang/grc/test_text.py
+++ /dev/null
@@ -1,23 +0,0 @@
-import pytest
-
-
-@pytest.mark.parametrize(
-    "text,match",
-    [
-        ("ι", True),
-        ("α", True),
-        ("ϟα", True),
-        ("ἑκατόν", True),
-        ("ἐνακόσια", True),
-        ("δισχίλια", True),
-        ("μύρια", True),
-        ("εἷς", True),
-        ("λόγος", False),
-        (",", False),
-        ("λβ", True),
-    ],
-)
-def test_lex_attrs_like_number(grc_tokenizer, text, match):
-    tokens = grc_tokenizer(text)
-    assert len(tokens) == 1
-    assert tokens[0].like_num == match
